executor/executor.go

```diff
@@ -122,22 +143,92 @@ func (e *Executor) Run(conflicts state.Keys, f func() error) {
 	e.tasks[id] = t
 
 	// Record dependencies
-	for k := range conflicts {
-		latest, ok := e.edges[k]
-		if ok {
-			lt := e.tasks[latest]
-			if !lt.executed {
-				if t.dependencies == nil {
-					t.dependencies = set.NewSet[int](defaultSetSize)
+	for k, v := range conflicts {
+		// Get last blocked transaction
+		parent, exists := e.edges[k]
```

Comment:
I don't love `parent` here given this `keyData` can be either the `parent` or some value back in the ancestry of keys.

Reply:
Maybe just `key`?